rllib/core/learner/learner_group.py

```diff
@@ -557,6 +551,197 @@ def _load_state(w):
+                for module_id, path in rl_module_ckpt_dirs.items():
+                    self._learner.module[module_id].load_state(
+                        path / RLMODULE_STATE_DIR_NAME
+                    )
+        else:
+            assert len(self._workers) == self._worker_manager.num_healthy_actors()
+            head_node_ip = ray.util.get_node_ip_address()
+
+            def _load_module_state(w):
+                # doing imports here since they might not be imported on the worker
+                import ray
+                import tempfile
+                import shutil
+
+                worker_node_ip = ray.util.get_node_ip_address()
+                # sync the checkpoints from the head to the worker if the worker is not
+                # on the same node as the head
+                tmp_marl_module_ckpt_dir = marl_module_ckpt_dir
+                tmp_rl_module_ckpt_dirs = rl_module_ckpt_dirs
+                if worker_node_ip != head_node_ip:
+                    if marl_module_ckpt_dir:
+                        tmp_marl_module_ckpt_dir = tempfile.mkdtemp()
+                        sync_dir_between_nodes(
+                            source_ip=head_node_ip,
+                            source_path=marl_module_ckpt_dir,
+                            target_ip=worker_node_ip,
+                            target_path=tmp_marl_module_ckpt_dir,
+                        )
+                    if rl_module_ckpt_dirs:
+                        tmp_rl_module_ckpt_dirs = {}
+                        for module_id, path in rl_module_ckpt_dirs.items():
+                            tmp_rl_module_ckpt_dirs[module_id] = tempfile.mkdtemp()
+                            sync_dir_between_nodes(
+                                source_ip=head_node_ip,
+                                source_path=path,
+                                target_ip=worker_node_ip,
+                                target_path=tmp_rl_module_ckpt_dirs[module_id],
+                            )
+                            tmp_rl_module_ckpt_dirs[module_id] = pathlib.Path(
+                                tmp_rl_module_ckpt_dirs[module_id]
+                            )
+                if tmp_marl_module_ckpt_dir and tmp_rl_module_ckpt_dirs:
+                    # If both a MARLModule checkpoint and RLModule checkpoints are
+                    # specified, load the MARLModule checkpoint first and then load
+                    # the RLModule checkpoints.
+                    w.module.load_state(
+                        tmp_marl_module_ckpt_dir, modules_to_load=modules_to_load
+                    )
+                    for module_id, path in tmp_rl_module_ckpt_dirs.items():
+                        w.module[module_id].load_state(path / RLMODULE_STATE_DIR_NAME)
+                elif tmp_marl_module_ckpt_dir:
+                    w.module.load_state(
+                        tmp_marl_module_ckpt_dir, modules_to_load=modules_to_load
+                    )
+                else:
```

Comment:
same comment here.
seems like you just need to try tmp_marl_module_ckpt_dir then tmp_rl_module_ckpt_dirs.

Reply:
fixed.